lib/checks/aria/valid-attr-value.js

```diff
@@ -1,26 +1,35 @@
 options = Array.isArray(options) ? options : [];
 
-var invalid = [],
-	aria = /^aria-/;
+const invalid = [];
+const aria = /^aria-/;
+const attrs = axe.utils.getNodeAttributes(node);
 
-var attr,
-	attrName,
-	attrs = axe.utils.getNodeAttributes(node);
+const skipAttrs = ['aria-errormessage'];
 
-var skipAttrs = ['aria-errormessage'];
+// aria-controls should only check if element exists if the element
+// doesn't have aria-expanded=false or aria-selected=false (tabs)
+// @see https://github.com/dequelabs/axe-core/issues/1463
+const preChecks = {
+	'aria-controls': function() {
+		return (
+			node.getAttribute('aria-expanded') !== 'false' &&
+			node.getAttribute('aria-selected') !== 'false'
+		);
+	}
+};
 
 for (var i = 0, l = attrs.length; i < l; i++) {
```

Comment:
While you're here and updating `var` to `const` already, maybe update this loop too?

```js
for (const attr of attrs) {
  const attrName = attr.name
  // ...
}
```

Reply:
I keep forgetting that `for of` is a thing.